src/components/FileViewer.jsx

```diff
@@ -12,17 +12,32 @@ class FileViewer extends React.Component {
   }
 
   componentDidMount () {
-    const intent = this.props.intent
-    cozy.client.intents.createService(intent, window)
-      .then(service => {
+    const { intentId } = this.props
+
+    let service, intent
+    cozy.client.intents.createService(intentId, window)
```

Comment:
Could you use async/await syntax in this function to let the code be more expressive?